### PR TITLE
Enabling looped execution for xdlops gemm

### PR DESCRIPTION
**Pull request by jerryyin**
This is second attempt after https://github.com/ROCmSoftwarePlatform/rocMLIR/pull/942. The loop iteration style will change to the following (according to CK style):

```
for m = 0 : mRepeat
 a = load(A)
  for n = 0 : nRepeat
   b = load(B)
    for k = 0 : K   
      c = a * b
```

Will run this through with tuning for the second time to confirm the issue bumped with broadcast is work-arounded.

### CODE REVIEW

File: mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp [hunk 1]
```diff
@@ -486,37 +475,28 @@ struct BlockwiseGemmV2RewritePattern
     // for(index_t n_i = 0; n_i < mRepeats; ++n_i)
     //   for(index_t k_i = 0; k_i < KPerThread; ++k_i)
     //       ldsToRegisterCopy[n_i, k_i]
-    auto outerLoopN = b.create<AffineForOp>(loc, 0, nRepeats);
+    auto outerLoopN = olmb.create<AffineForOp>(loc, 0, nRepeats);
     auto olnb = ConversionPatternRewriter::atBlockBegin(outerLoopN.getBody(),
-                                                        b.getListener());
+                                                        olmb.getListener());
     auto innerLoopNK = olnb.create<AffineForOp>(loc, 0, KPerThread);
```
Comment (giuseros) [inline mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: Why not moving this inside `ldsToRegisterCopy`? In this way you don't have to duplicate the code for loading K

Comment (jerryyin) [reply mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: Yah good idea. Will do

Comment (jerryyin) [reply mlir/lib/Dialect/Rock/Transforms/BlockwiseGemmToThreadwise.cpp]: Done in 0477f189913225e020d40be0c143dc002c9ed406

**Comment by jerryyin** [comment]
Performance report is here: http://rocmhead:8080/job/MLIR/job/mlir/job/PR-976/8/Performance_20report_20for_20gfx90a/

Minor negative impact on fwd and bwd path but 4% ~ 14% improvement on wrw cases.

**Review by giuseros — APPROVED** [review]
LGTM

**Comment by giuseros** [comment]
Hi @jerryyin , any reason why not merging this in?

**Comment by jerryyin** [comment]
@giuseros I was waiting a full weekly pass before this one... A little bit concerned about non weekly run after upstream merge attempt. In case anything happens, I don't want this PR to get into the way. But you are right, I might be too cautious here and affected by previous compiler backend issue.

**Pull request merged**